forc-pkg/src/pkg.rs

```diff
@@ -1587,6 +1587,9 @@ pub fn compile(
     namespace: namespace::Module,
     source_map: &mut SourceMap,
 ) -> Result<(Compiled, Option<namespace::Root>)> {
+    //TODO remove once type engine isn't global anymore.
+    sway_core::clear_lazy_statics();
```

Comment:
Ahhh this should be at the beginning of `build`, not `compile` - otherwise we clear all the type state of dependency packages.